src/App.jsx

```diff
@@ -1,19 +1,32 @@
 import React from "react";
 import Header from "./components/Header/Header";
 import Footer from "./components/Footer/Footer";
-import Main from "./components/Main/Main";
 import ErrorBoundary from "./components/ErrorBoundary/ErrorBoundary";
 import ModalManager from "./components/Modal/ModalManager";
+import {
+    BrowserRouter as Router,
+    Route,
+    Switch,
+} from "react-router-dom";
+import Home from "./pages/Home";
+import NoMatch from "./pages/NoMatch";
+import MovieDetails from "./pages/MovieDetails";
+import SearchResult from "./pages/SearchResult";
 
 export default function App() {
-    const [movie, setMovie] = React.useState(null);
-
     return(
-        <ErrorBoundary>
-            <Header />
-            <Main />
-            <Footer />
-            <ModalManager />
-        </ErrorBoundary>
+        <Router>
+            <ErrorBoundary>
+                <Header />
+                <Switch>
+                    <Route exact path="/" component={Home} />
+                    <Route path="/film/:id" component={MovieDetails} />
+                    <Route path="/search/:searchQuery" component={SearchResult} />
```

Comment:
to think: from my point of view `/` and `/search/:searchQuery` pages share same logic (search bar at the top and list of movies at the bottom). Could one component be used for two routes? What do you think? 🤔

Reply:
fixed